packages/yoastseo/src/scoring/assessments/seo/ImageAltTagsAssessment.js

```diff
@@ -59,46 +63,43 @@ export default class ImageAltTagsAssessment extends Assessment {
 		return assessmentResult;
 	}
 
-	/**
-	 * Checks whether the paper has text with at least 1 image.
-	 *
-	 * @param {Paper}       paper       The paper to use for the assessment.
-	 * @param {Researcher}  researcher  The Researcher object containing all available researches.
-	 *
-	 * @returns {boolean} True when there is text.
-	 */
-	isApplicable( paper, researcher ) {
-		this.imageCount = researcher.getResearch( "imageCount" );
-		return paper.hasText() && this.imageCount > 0;
-	}
-
 	/**
 	 * Calculates the result based on the availability of images in the text.
 	 *
-	 * @returns {Object} The calculated result.
+	 * @param {Paper} paper The Paper object to assess.
+	 *
+	 * @returns {{score: number, resultText: string}} The calculated result.
 	 */
-	calculateResult() {
-		// The number of images with no alt tags.
+	calculateResult( paper ) {
+		// The number of images with no alt attributes.
 		const imagesNoAlt = this.altTagsProperties.noAlt;
-		const { good: goodResultText, noneHasAltBad, someHaveAltBad } = this.getFeedbackStrings();
+		const { good: goodResultText,  noImagesBad, noneHasAltBad, someHaveAltBad } = this.getFeedbackStrings();
 
-		// None of the images has alt tags.
+		// There are no images or no text
+		if ( this.imageCount === 0 || paper.getText().length === 0 ) {
+			return {
+				score: this._config.scores.bad,
+				resultText: noImagesBad,
+			};
+		}
```

Comment:
I think it's enough to check only for no images. We don't want to return a bad score if there are images but no text, right? Though apparently `paper.getText().length === 0 ` returns `false` when there are images but no text, but still, the check seems redundant if we are already checking if there are no images.

Reply:
Agree!